src/FixedDataTable.js

```diff
@@ -959,7 +999,8 @@ var FixedDataTable = createReactClass({
     }
 
     var lastScrollToRow  = oldState ? oldState.scrollToRow : undefined;
-    if (props.scrollToRow != null && (props.scrollToRow !== lastScrollToRow || viewportHeight !== oldViewportHeight)) {
+    var rowHeightChanged = props.scrollToRow != null ? this._scrollHelper.getRowHeightChanged(props.scrollToRow) : 0
```

Comment:
if we have less data rows, like 10 rows, expand the last two rows, then collapse the last row, the last row also will not be positioned at the end, FYI